### ISSUE TITLE
Additional checkout fields: Integrate the visiblity rules parsers into additonal fields.

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#46689]
Blocked by https://github.com/woocommerce/woocommerce/issues/46685 and https://github.com/woocommerce/woocommerce/issues/46687, once we have the parsers in place, we should integrate them into additional fields, this includes:

- Adding a visibility rules field when registering additional fields.
- Validating them before adding the fields to the page (after all fields are registered).
- Reading them in frontend.
- Initiating the JS parser so we can pass in Cart, Customer, Address, and Order data and decide what to show or not. I envision this happening on some level after building the fields array but before rendering it in Form component.
- Deciding if validation and persistent and sanitation should be skipped in Store API based on that as well.

**Comment by senadir** [comment]
Closing this in favor of new set of issues
